src/use-force-update.ts

```diff
@@ -1,21 +1,14 @@
-import { useState } from 'react';
+import { useReducer } from 'react';
 
-interface VoidFunction {
-  (): void;
-}
+type VoidFunction = () => void;
+type VoidFunctionCreator = () => VoidFunction;
+type ToggleReducer = (state: boolean, action: void) => boolean;
 
-interface VoidFunctionCreator {
-  (): VoidFunction;
-}
+const reducer: ToggleReducer = state => !state;
 
-const toggle = (state: boolean): boolean => !state;
-
-const useForceUpdate: VoidFunctionCreator = (): VoidFunction => {
-  const [ , setState ] = useState<boolean>(true);
-  const forceUpdate: VoidFunction = (): void => {
-    setState(toggle);
-  };
-  return forceUpdate;
+const useForceUpdate: VoidFunctionCreator = () => {
+  const [, dispatch] = useReducer(reducer, true);
+  return dispatch as VoidFunction;
```

Comment:
will wait on feedback regarding the `as VoidFunction`, maybe there's a way around it.

no hurries to merge, it's not even a direct dependency of mine :P
I stumbled upon it while reading your [`use-react-router`](https://github.com/CharlesStover/use-react-router).